source/client-side-operations-timeout/tests/pending-response.yml

```diff
@@ -0,0 +1,349 @@
+          - connectionCheckedInEvent: {}
+  # Attempting a pending response read on a realistic timeout operation that can
+  # immediately read from the TCP buffer should complete the pending read and
+  # the connection should be checked out.
+  - description: "timeout op with response and no error"
+    operations:
+      # Create a failpoint to block the first operation.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: {times: 1}
+            data:
+              failCommands: ["insert"]
+              blockConnection: true
+              blockTimeMS: 250
+      # Execute operation with timeout less than block time.
+      - name: insertOne
+        object: *collection
+        arguments:
+          timeoutMS: 75
+          document: {_id: 3, x: 1}
+        expectError:
+          isTimeoutError: true
+      # Execute a subsequent operation to complete the read.
+      - name: find
+        object: *collection
+        arguments:
+          timeoutMS: 200
+          filter: {_id: 1}
+    expectEvents:
+      - client: *client
+        events:
+          - commandFailedEvent:
+              commandName: insert
+          - commandSucceededEvent:
+              commandName: find
+      - client: *client
+        eventType: cmap
+        events:
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Insert fails.
+          - connectionPendingResponseStarted: {}
+          - connectionPendingResponseSucceeded: {}
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Find succeeds.
+  # It may take multiple calls to the pending response handler to drain the
+  # inbound buffer.
+  - description: "multiple calls to drain buffer"
+    operations:
+      # Create a failpoint to block the first and second operation.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: {times: 1}
+            data:
+              failCommands: ["insert"]
+              blockConnection: true
+              blockTimeMS: 500
+      # Execute operation with timeout less than block time.
+      - name: insertOne
+        object: *collection
+        arguments:
+          timeoutMS: 50
+          document: {_id: 3, x: 1}
+        expectError:
+          isTimeoutError: true
+      # Execute a subsequent operation with a timeout less than the block time.
+      - name: find
+        object: *collection
+        arguments:
+          timeoutMS: 50
+          filter: {_id: 1}
+        expectError:
+          isTimeoutError: true
+      # Execute a final operation to drain the buffer.
+      - name: find
+        object: *collection
+        arguments:
+          filter: {_id: 1}
```

Comment:
Are we missing an explicit `timeoutMS` value here?

Reply:
If we do not have timeout we fallback to sockettimeout, but you are right, we better to have the explicit timeout here.